### ISSUE TITLE
Creality CR10-S printer is failing with "thermal runaway"

### ISSUE THREAD

**Issue by ajlennon** [DoESLiverpool/somebody-should#1163]
The bed is heating and it seems to be reading a temperature increase.

But after a while the system halts with a "thermal runwaway" error message.

Also the heating bed connector at the rear of the main PSU box is getting very hot.

Others have apparently seen this problem

https://www.reddit.com/r/CR10/comments/8a6s05/is_it_normal_for_the_bed_cable_connector_to_get/

I've contacted the vendor for a response

**Comment by Thingomy** [comment]
Hot bed connector was found to be underrated (either 5A or 1A) when it was taking 10A. It turns out the connector was getting hot enough to melt it's solder, and was being held together by heatshrink. Connector has been replaced with bullet connectors.

Box was reassembled and machine was fired up, whereupon a clunk was heard from inside the box, this turned out to be the hotbed mosfet falling off due to overheating melting it's solder. It was soldered back on on the offchance it was a one off; it wasn't a one off. I'm 95% sure there is no short in the hotbed.

Machine is currently in pieces awaiting either somone that knows thier way around mosfets to prod it, or me to bring in a bunch of additional tools.

**Comment by zarino** [comment]
When I hear the word "mosfet" I immediately think @magman2112 – pinging him here in case he has any expertise to lend!

**Comment by ajlennon** [comment]
Hehe. Yes @magman2112 was involved with @Thingomy in the investigation. 

I've had a response from Ristar, whom I bought the printer from via Amazon

>  Dear Customer,
>
>Thanks for your contact.
>We apologize for the case you meet.
>
>Could you please provide us more specific information about the problem?
>you can also take some photos or a short to show it, then our technical will specify the problem. 
>And we will according to the reasons do the appropriate treatment.  We appreciate your cooperation.
>
>Looking forward to your reply.
>
>Kind regards 

My response:

>Hi,
>
>Thanks for your response. We've done some more investigation here and it appears as though you are >putting more current through the heating bed connector than its specification allows.
>
>
>Notes:
>
>"Hot bed connector was found to be underrated (either 5A or 1A) when it was taking 10A. It turns out >the connector was getting hot enough to melt it's solder, and was being held together by heatshrink. ... >Machine was fired up, whereupon a clunk was heard from inside the box, this turned out to be the >hotbed mosfet falling off due to overheating melting it's solder. It was soldered back on on the >offchance it was a one off; it wasn't a one off. I'm 95% sure there is no short in the hotbed."
>
>I am not in the office today to provide pictures but the issue appears the same as on this page. So it >looks like others are seeing this problem too.
>
>https://www.reddit.com/r/CR10/comments/8a6s05/is_it_normal_for_the_bed_cable_connector_to_get/
>
>I like this printer a lot and have been recommending it to colleagues in my workspace, but I am >concerned about the issues above. Please help to resolve them so I can continue to recommend you >guys!
>
>Thanks,
>
>Alex Lennon

![image](https://user-images.githubusercontent.com/1537834/60421974-d9c52180-9be2-11e9-9485-778b51a20e9f.png)

Will keep this issue updated with their response

**Comment by DocTrucker** [comment]
I'd mentioned about the hazards  the Anet printers on the DoES Liverpool google mailing lists last december and the Creality machines and most at the same price break are very similar. Even the much more expensive Up Printer has very small CSA wire for the heaters.

The creality machines at least used to ship with a version of Marlin with thermal protections disabled, thankfully that has either changed or you caught it.

It's note worthy that the Ultimakers are one of the only suppliers to say their machine can run unattended on the CE certification.

All that said I'm working on the safeties on my 'small holding' at home to save any potential grumbles from the insurance company. Passive saftey (ie heaters at 100% duty never reach a dangerous temperature) is first, but is generally frustrating. Firmware safteys are the next option, and finally thermal fuses switches and fuses mounted on the hot zones can save your building if it all goes belly up.

As with most electronics it's the negative leg that is modulated for heater control so you can feed the posative side of the heaters seperately to the rest of the board (tied negative poles) and interlock that PSU on faults. Most boards have a PS_ON pin which can be used to drop relays. If you go down that route remember SSRs and MOSFETs are not safety devices.

**Comment by jrp** [comment]
I'm planning to be in tonight with the right tools to to do some more diagnostics on this.

**Comment by DocTrucker** [comment]
Expect I may be stating the obvious to you but have a good look over the pluggable  heater contacts for signs of arking and check gauge of wire against the expected current draw. If the creality is the machine I'm thinking of it's got (or previous models had) an aluminium heatbed. It's worth ditching the connector and soldering wire (maybe new) directly to the bed and fully stress shield the joint. If you do this preheating the aluminium plate on a gas stove helped me. I needed a gas powered soldering iron to get enough heat to work with high temp (silicone insulated) 14awg, but that is potentially overkill for a standard heatbed. Check thermistors are secure too where they are removable.

**Comment by Thingomy** [comment]
I had a look tonight at the mosfet arrangment to switch the heated bed, it is an off the shelf board that effectively emulates a relay. It is officialy goosed. That tends to happen when parts get hot enough to melt off. I replaced the failed mosfet with a much lower rated one for the purpose of testing whether the board worked, and it still does nothing. 

So we need a replacement for this board.

As for how it died, I'm puzzled. The Mosfet is actually well over rated, (98A is mentioned in the datasheet, which is about what I'd spec for driving a 10A load), and the connectors and wires in the area look up to spec too (other than the one that started all this). It's possible I did something to it during the other fix, but I can't work out what. I even confirmed I never reversed it's power supply, which would have had this effect.

I'm now unavailable for a few weeks, so I'm going to have to hand this one off to the next interested party.

PS. I left a bag of mosfets in the area, could someone either keep them with the machine, or stick them in my box please?

PPS. I got as far as to work out how the mosfet overheated so fast. It failed mostly conductive, at 0.18ohms (including some of the wiring) so as soon as the PSU came on, it imediately outputs a measured 7.8amps to the hotbed; and so runs at 13ish watts, and the heatsink fails to disapate it. It's possible that during it's failing process it's resistance was higher, producing more heat.

**Comment by DocTrucker** [comment]
Failed closed circuit is a common failure mode of MOSFETs & SSRs. This is
why Omron phorbid the use of their devices in safety  critical circuits
where there is no backup to stop the flow of current in the event of
failure.

While they are great for high frequency pwm without having to worry about
inductive kick back from the coil, or electrical noise from contact arking,
they are crap at dealing with transient voltages that mechanical relays
hardly notice.

In automotive use their circuits tend to be built with TVS (transient
voltage suppression) diodes and other over voltage and crowbar like
protection devices. Designing the protection isn't straight forward as you
have to have some idea of the spike that you're protecting from. In the
DoES case it could have been created on the device, in it's power supply,
come from the mains, lightening within a mile or so, or someone else in
DoES recent creation maybe accidentally very electrically noisey. ...or
just a poor quality MOSFET board.

Long story short buy your SSRs or MOSFETs boards from reputable sources,
ensure there is an additional mechanical relay to kill power to the heaters
(on dc or ac side of heater psu), and consider a thermal fuses.

**Comment by ajlennon** [comment]
Ristar haven't come back to me.

Have written them a bad review and requested a refund. Will see what happens...

**Comment by Sean-anotherone** [comment]
https://www.aliexpress.com/wholesale?catId=0&initiative_id=SB_20190729115336&SearchText=creality+board

**Comment by ajlennon** [comment]
Yeah we might end up going that way @Sean-anotherone. Can you recommend a board for the Creality CR-10S ?

**Comment by DocTrucker** [comment]
https://forum.duet3d.com/topic/10542/converting-creality-cr-10-s5-to-duet-wifi-24v

**Comment by ajlennon** [comment]
Still no response from Ristar so escalated to Amazon. They have agreed a full refund on return of the Creality printer so it needs packing up for a pickup

**Comment by ajlennon** [comment]
Say goodbye 
![image](https://user-images.githubusercontent.com/1537834/62822948-e4c88580-bb82-11e9-98ec-576bd21aeb9d.jpg)

**Comment by ajlennon** [comment]
The failed unit should be picked up today. I've ordered a replacement (same model) which should be with us by end of week.

**Comment by ajlennon** [comment]
OK the replacement Creality CR10-S has arrived @jackie1050 
 
I have

- put it all together
- updated the firmware to Marlin 1.9
- had a rough go at levelling the bed
- connected up OctoPi, checked it worked and is heating up the heat/bed
- put the filament in.

I'm just doing a test print but feel free to have a go at bed levelling and some prints yourself.

I'll be in tomorrow with a bit of luck to chase up why Amazon have still not picked up the other unit and to move things around to take this one off the workspace desk...

**Comment by ajlennon** [comment]
Had a quick go at printing. Seemed to be going well then my test cube came unstuck. Base probably just needs a good clean

**Comment by jackie1050** [comment]
Sounds good - I won't be in until tomorrow about 1pm but looking forward to trying it out.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by RussCoty** [comment]
Looks good. Is it going to live on that worktop where the other one was? Or is there room near the ultimakers?

**Comment by jackie1050** [comment]
The space near the Ultimakers is really a desk - we're aiming to return it to that use once we've finished the large filament reel that won't fit in the usual space (and once Felicity's fixed).
So probably it will go where the other one was - unless there's a better plan?

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by ajlennon** [comment]
Good plan. The setup went better than I expected. I’m quite annoyed Amazon haven’t picked up the original one yet. I’ve been onto them a number of times. Will pursue them again tomorrow and move the pick up package somewhere, then move the new one to where the original one was and we can get back to 3D printing :)

**Comment by jackie1050** [comment]
If it's staying at DoES you should give it a name

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by ajlennon** [comment]
Oooh yes. Well hopefully this won’t blow up. I meant to say let’s keep checking the heat on the connectors.

So what ideas do people have for the new printer? Throwing it open...

**Comment by ajlennon** [comment]
![CAD66DAF-67A6-4BEB-BBF7-6B3B935ED92D](https://user-images.githubusercontent.com/1537834/63210315-88b2b380-c0e4-11e9-97ff-4d8b3452bb6d.jpeg)

**Comment by ajlennon** [comment]
Checked the back connectors and they’re not hot at all

**Comment by jrp** [comment]
Could I bother you for a picture of the offending connectors on the new machine? I'm intregued as to how they have redesigned it.

JR

**Comment by ajlennon** [comment]
Looks same as before. Nice and cool at the moment 

![735FC47D-1C44-41A1-BB03-FBCBBA2B1934](https://user-images.githubusercontent.com/1537834/63214802-e0691300-c114-11e9-96f5-aa018dab5f41.jpeg)

**Comment by ajlennon** [comment]
So a bit more of an update from yesterday's attempts to print.

I had a lot of failures. Mainly the plastic wasn't sticking to the glass surface. I tried cleaning it and I tried a few different types of glue.

I wonder if the head is dragging against the plastic at the top.

I did get one "boaty" print which was really really good so it's possible but it's just a bit temperatmental at the moment.

Perhaps those who are more expert in these matters @Sean-anotherone or @jrp could cast their eye over it...?

**Comment by DocTrucker** [comment]
Manually level the bed as best you can (system doesn't need to be on) to
give the probe based bed level correction a fighting chance.

A clean glass bed (IPA, Ethanol, or Acetone should be fine) should be all
you need for PLA, but basic, cheap hair strong hair spray may give you
a.little more of a fighting chance.

Heat the bed upto temperture if required and leave for half an hoir or so.
Run a bed probe and store the correction mesh. Not sure how to view the bed
level correction results on Marlin, but check to ensure it doesn't show any
evidence of twist about the y axis. Ideally you'd be best off manually
levelling the bed to the point where bed probing isn't required, but the
machine may not be stable enough to do that.

CAD the largest x and y dimension cube you can. Slice it with a a grid fill
density low enough to give 10-20mm spacing of the hatch lines. Print only
the first layer and look at the width of the lines over the bed. Slight
variations or errors will bee shown up quite clearly by the lines becoming
thicker or thinner. If all consistant and your wall thickness for a single
wall cube is on target then it's something a little more in depth.

**Comment by irenge** [comment]
Thanks!

- the old printer is now gone.

- looks like the filament needed to be printing hotter for starters

- @jackie1050 I've just found out we can print directly from Cura to the Printer via OctoPi

![Screenshot from 2019-08-19 12-30-18](https://user-images.githubusercontent.com/1784976/63262213-4b355e00-c27d-11e9-8ffa-b53484d9e464.png)

**Comment by ajlennon** [comment]
(oops - bit of confusion - that was me :) )

**Comment by ajlennon** [comment]
ok with a lot of help from JC I think we're getting there.

- it's really easy on the CR-10S UI to twiddle the knob on the info screen and that changes the feed rate. I bet I had done this by accident.

- bed levelling has been done with a post-it note and then I've mesh levelled everything to zero for a start

- the advice is to use no glue. wash down the glass with warm water, dry, then clean to "squeakiness" with glass cleaner.

- then print a grid of 1cm calibration cubes in a 3x3 arragement over the entire grid to check levelling.

I'll start building up a page on the Wiki to document this stuff.

For now I'll close down this issue...

Thanks for all the help everybody!

**Comment by DoESsean** [comment]
I would like to suggest the name Ramon.

There will be a small prize if anyone can figure out the convoluted way I arrived at that suggestion by the 27th of August.

**Comment by ajlennon** [comment]
Ramen?

**Comment by DocTrucker** [comment]
Quite clearly because you wish it had more RAM on it... ;-)

**Comment by DoESsean** [comment]
Those are both one step names, nowhere near convoluted enough. This is a
three step name.

**Comment by MatthewCroughan** [comment]
Taking over the testing on this now, since the first layer of large prints fail. Will post my results to the wiki, with prefix "Matt:" prior to the statement. We can then concatenate it.

**Comment by ajlennon** [comment]
Ok can we (you) break this out into a new setup issue as I think we’ve addressed the meat of what this one was about
